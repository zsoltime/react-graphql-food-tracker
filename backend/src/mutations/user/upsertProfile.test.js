'use strict';

const upsertProfile = require('./upsertProfile');

describe('upsertProfile mutation', () => {
  it('throws error if user is not logged in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      upsertProfile(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('creates profile if user has none', async () => {
    const args = { name: 'Zsolt', height: 198 };
    const mockedContext = {
      db: {
        mutation: {
          createProfile: jest.fn(),
        },
        query: {
          profiles: jest.fn().mockImplementation(() => []),
        },
      },
      request: { userId: 'testUserId' },
    };

    await upsertProfile(null, args, mockedContext);

    expect(
      mockedContext.db.mutation.createProfile
    ).toHaveBeenCalled();
    expect(
      mockedContext.db.mutation.createProfile.mock.calls[0]
    ).toMatchSnapshot();
  });

  it('updates profile if user has one', async () => {
    const args = { name: 'Zsolt', height: 198 };
    const mockedContext = {
      db: {
        mutation: {
          updateProfile: jest.fn(),
        },
        query: {
          profiles: jest
            .fn()
            .mockImplementation(() => [{ id: 'testProfileId' }]),
        },
      },
      request: { userId: 'testUserId' },
    };

    await upsertProfile(null, args, mockedContext);

    expect(
      mockedContext.db.mutation.updateProfile
    ).toHaveBeenCalled();
    expect(
      mockedContext.db.mutation.updateProfile.mock.calls[0]
    ).toMatchSnapshot();
  });
});
