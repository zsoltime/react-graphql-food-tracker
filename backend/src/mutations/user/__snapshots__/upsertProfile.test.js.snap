// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upsertProfile mutation creates profile if user has none 1`] = `
Array [
  Object {
    "data": Object {
      "height": 198,
      "name": "Zsolt",
      "user": Object {
        "connect": Object {
          "id": "testUserId",
        },
      },
    },
  },
  undefined,
]
`;

exports[`upsertProfile mutation throws error if user is not logged in 1`] = `"You must be signed in to update your profile"`;

exports[`upsertProfile mutation updates profile if user has one 1`] = `
Array [
  Object {
    "data": Object {
      "height": 198,
      "name": "Zsolt",
    },
    "where": Object {
      "id": "testProfileId",
    },
  },
  undefined,
]
`;
